subsys/dfu/boot/mcuboot.c

```diff
@@ -185,21 +185,30 @@ int boot_request_upgrade_multi(int image_index, int permanent)
 
 bool boot_is_img_confirmed(void)
 {
+	struct boot_swap_state state;
 	const struct flash_area *fa;
 	int rc;
-	uint8_t flag_val;
 
 	rc = flash_area_open(FLASH_AREA_IMAGE_PRIMARY, &fa);
 	if (rc) {
 		return false;
 	}
 
-	rc = boot_read_image_ok(fa, &flag_val);
-	if (rc) {
+	rc = boot_read_swap_state(fap, &state);
+	if (rc != 0) {
 		return false;
 	}
 
-	return flag_val == BOOT_FLAG_SET;
+	if (state.magic == BOOT_MAGIC_UNSET) {
```

Comment:
It's been a long time since I configured mcuboot with this option, but how does this work with the optional Kconfig for a provisioning image, or bootstrap image? I can't remember much more than the name

Reply:
This is related this ticket https://github.com/mcu-tools/mcuboot/issues/1066
An image I referred is the image which was flashed directly to the primary slot using programing interface (like JLink).